source/Weapon.cpp

```diff
@@ -17,21 +96,121 @@ void Weapon::onPose(Array<shared_ptr<Surface> >& surface) {
 	}
 }
 
+void Weapon::simulateProjectiles(SimTime sdt, const Array<shared_ptr<TargetEntity>>& targets, const Array<shared_ptr<Entity>>& dontHit) {
+	// Iterate through projectiles for hit/miss detection here
+	for (int p = 0; p < m_projectiles.size(); p++) {
+		shared_ptr<Projectile> projectile = m_projectiles[p];
+		projectile->onSimulation(sdt);
+		// Remove the projectile for timeout
+		if (projectile->remainingTime() <= 0) {
+			// Expire
+			m_scene->removeEntity(projectile->name());
+			m_projectiles.remove(p);
+			--p;
+		}
+		else if (!m_config->hitScan) {
+			// Distance at which to delcare a hit
+			const float hitThreshold = m_config->bulletSpeed * 2.0f * (float)sdt;
+			// Look for collision with the targets
+			const Ray ray = projectile->getCollisionRay();
+			float closest = finf();
+			Model::HitInfo info;
+			shared_ptr<TargetEntity> closestTarget;
+			for (shared_ptr<TargetEntity> t : targets) {
+				if (t->intersect(ray, closest, info)) {
+					closestTarget = t;
+				}
+			}
+			// Check for target hit
+			if (closest < hitThreshold) {
+				m_hitCallback(closestTarget);
+				// Offset position slightly along normal to avoid Z-fighting the target
+				drawDecal(info.point + 0.01 * info.normal, m_camera->frame().lookVector(), true);
+				projectile->clearRemainingTime();
+			}
+			// Handle (miss) decals here
+			else {
+				Array<shared_ptr<Entity>> dontHitItems = dontHit;
+				dontHitItems.append(m_currentMissDecals);
+				dontHitItems.append(targets);
```

Comment:
avoiding hitting targets seems a little weird here since a miss should have missed the targets in the first place. It probably isn't a problem to skip them, but they also shouldn't be hit here either since it's supposed to be a miss case.

Reply:
This is probably correct, I just don't understand the context of what is being hit or missed here and why we're trying to avoid some things in the intersect call. Maybe this just means we need better comments describing these situations in more detail?

Reply:
I don't think we "need" to add the targets to the don't hit list here (we are just getting the normal for the miss decal placement) we assume targets aren't something we need to worry about at this stage.

Comments could definitely be improved.